packages/clay-management-toolbar/src/ClayManagementToolbar.soy

```diff
@@ -230,17 +242,93 @@
 				</li>
 				<li class="nav-item">
 					<span class="navbar-text">
-						{msg desc="Says how many items of the total are selected"}
-							{$selectedItems}-of-{$totalItems}
-						{/msg}
-						{sp}
-						<span class="navbar-breakpoint-down-d-none">
-							{msg desc="Items selected"}
-								items-selected
+						{if $selectedItems == $totalItems}
+							{msg desc="All selected"}
+								all-selected
 							{/msg}
-						</span>
+						{else}
+							{msg desc="Says how many items of the total are selected"}
+								{$selectedItems}-of-{$totalItems}
+							{/msg}
+							{sp}
+							<span class="navbar-breakpoint-down-d-none">
+								{msg desc="Items selected"}
+									items-selected
+								{/msg}
+							</span>
+						{/if}
 					</span>
 				</li>
+				<li class="nav-item nav-item-shrink">
+					{if $clearSelectionURL}
+						{call ClayLink.render}
+							{param elementClasses: 'nav-link' /}
+							{param href: $clearSelectionURL /}
+							{param label kind="html"}
+								<span class="text-truncate-inline">
+									<span class="text-truncate">
+										{msg desc="Clear"}
+											clear
+										{/msg}
+									</span>
+								</span>
+							{/param}
+						{/call}
+					{else}
+						{call ClayButton.render}
+							{param elementClasses: 'nav-link' /}
+							{param events: ['click': $_handleClearSelectionButtonClick] /}
+							{param label kind="html"}
+								<span class="text-truncate-inline">
+									<span class="text-truncate">
+										{msg desc="Clear"}
+											clear
+										{/msg}
+									</span>
+								</span>
+							{/param}
+							{param ref: 'clearSelectionButton' /}
+							{param style: 'unstyled' /}
+							{param type: 'button' /}
+						{/call}
+					{/if}
+				</li>
+				{if $selectedItems == $totalItems}
```

Comment:
hey @carloslancha I'm a bit confused with this, if all the items are already selected, we will show the `Select all` button. Is this expected?